### PR TITLE
Decision making statements / it-academy.by / JC1-110-22 / Alena Trafimovich / Done tasks 1-14

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 1]
```diff
@@ -37,6 +37,22 @@ public static void main(String[] args) {
...
+                result = "Good day";
+            } else if (hour < 24) {
+                result = "Good evening";
+            } else {
+                result = null;
+            }
+        } else {
+            result = null;
+        }
+        return result;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java:56]: this is hard

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task07.java [hunk 2]
```diff
@@ -38,6 +38,12 @@ public static void main(String[] args) {
     static boolean isLeapYear(int year) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return false;
+        boolean result;
+        if (year % 400 == 0) {
+            result = true;
+        } else if (year % 100 == 0) {
+            result = false;
+        } else result = (year % 4 == 0);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task07.java:46]: bad practice

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package com.rakovets.course.java.core.practice.decision_making_statements;
 
+import java.util.Objects;
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java:4]: What is it?

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java [hunk 4]
```diff
@@ -40,6 +42,14 @@ public static void main(String[] args) {
...
-        return 0;
+        int result = damage;
+        if (hasHolyAttribute) {
+            if ("SAINT".equals(typeMob)) {
+                result = damage / 2;
+            } else if ("UNDEAD".equals(typeMob) || "ZOMBIE".equals(typeMob)) {
+                result = (int) (damage * 1.5);
+            }
+        }
+        return result;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java:53]: use switch for it

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java [hunk 5]
```diff
@@ -45,6 +45,34 @@ public static void main(String[] args) {
     static String getZodiacSign(int day, int month) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        String result;
+        if (month < 1 || day < 1 || month > 12 || day > 31) {
+            result = null;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java:50]: This is useless

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java [hunk 6]
```diff
@@ -38,12 +38,38 @@ public static void main(String[] args) {
...
+        } else if ((2009 - year) % 12 == 0) {
+            result = "Ox";
+        } else if ((2010 - year) % 12 == 0) {
+            result = "Tiger";
+        } else if ((2011 - year) % 12 == 0) {
+            result = "Rabbit";
+        } else {
+            result = "Dragon";
+        }
+        return result;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java:73]: this is hard

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Review by rakovets — APPROVED** [review]
Approved

**Comment by AlenaTrafimovich** [comment]
I wanted to add some more improvements, but it's closed already:))

**Pull request closed without merging**